californium-core/src/test/java/org/eclipse/californium/core/coap/RequestTest.java

```diff
@@ -67,8 +68,21 @@ public void testSetURISetsDestination() throws UnknownHostException {
 
 	@Test
 	public void testSetURISetsUriHostOption() throws UnknownHostException {
-		Request req = Request.newGet().setURI("coap://iot.eclipse.org");
-		assertThat(req.getOptions().getUriHost(), is("iot.eclipse.org"));
```

Comment:
Regarding the problem this issue is all about: the host option only needs to be set in a `Request` if it is *not* the same literal IP address as the Request's destination address. The way I read this is that the usual case is that the URI contains a literal IP address. If not, we simply put the hostname into the host option (no need to resolve it).